### PR TITLE
feat(lucene): add min attribute for boolean query

### PR DESCRIPTION
**Pull request by duncdrum**
see eXist-db/exist#3107

### CODE REVIEW

File: src/main/xar-resources/data/lucene/lucene.xml [hunk 1]
```diff
@@ -621,6 +620,8 @@
                         </variablelist>
                         <para>For instance:</para>
                         <programlisting language="xquery" xlink:href="listings/listing-41.txt"/>
+                        <para>To optimize performance you can also specify a minium number of matches to cut-off needless disjunctive searches, using the <literal>min</literal> attribute:</para>
```
Comment (dizzzz) [inline src/main/xar-resources/data/lucene/lucene.xml]: only for the "should" modus

Comment (joewiz) [reply src/main/xar-resources/data/lucene/lucene.xml]: Also, typo: `minium`

Comment (joewiz) [reply src/main/xar-resources/data/lucene/lucene.xml]: @dizzzz So min only works as an attribute on `<bool>` when ALL of its child elements have the `@occur="should"` attribute? That’s not mentioned in https://github.com/eXist-db/exist/issues/392, where the `@occur` attribute isn’t on any of the child elements. In that case of omitted `@occur` attributes, are they implied?

Comment (duncdrum) [reply src/main/xar-resources/data/lucene/lucene.xml]: Yes I’m confused about the should thing too now.

Comment (dizzzz) [reply src/main/xar-resources/data/lucene/lucene.xml]: it works on the `bool` element, eg `<bool min="4">` as in the test

Comment (duncdrum) [reply src/main/xar-resources/data/lucene/lucene.xml]: so it wouldn't work with `must` or `not` ?

Comment (dizzzz) [reply src/main/xar-resources/data/lucene/lucene.xml]: correct (according to the Lucene javadoc)

Comment (dizzzz) [reply src/main/xar-resources/data/lucene/lucene.xml]: refs:
javadoc: https://lucene.apache.org/core/3_0_3/api/core/org/apache/lucene/search/BooleanQuery.html#setMinimumNumberShouldMatch(int) 
code: https://www.cc.gatech.edu/computing/pag/tmp/libanalysis/typestate/lusearch/org/apache/lucene/search/BooleanQuery.java.html

Comment (dizzzz) [reply src/main/xar-resources/data/lucene/lucene.xml]: image

Comment (joewiz) [reply src/main/xar-resources/data/lucene/lucene.xml]: @dizzzz The link to the javadocs is helpful. Here's the link to the javadocs corresponding to the version we're currently using in eXist, Lucene 4.10.4: https://lucene.apache.org/core/4_10_4/core/org/apache/lucene/search/BooleanQuery.html#setMinimumNumberShouldMatch(int)

Comment (duncdrum) [reply src/main/xar-resources/data/lucene/lucene.xml]: ok i ll add that as not to this PR to make it more clear.

**Pull request merged**